src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -280,6 +284,41 @@ private static Glyph GetGlyph(SymbolKind kind, Accessibility? declaredAccessibil
             return result.ToImmutable();
         }
 
+        private static void AddNamesFromExistingOverloads(CSharpSyntaxContext context, ArrayBuilder<(string name, SymbolKind kind)> result, CancellationToken cancellationToken)
+        {
+            var parameterSyntax = context.LeftToken.GetAncestor(n => n.IsKind(SyntaxKind.Parameter)) as ParameterSyntax;
+            if (parameterSyntax is not { Type: { } parameterType, Parent: { Parent: MethodDeclarationSyntax method } })
+            {
+                return;
+            }
+
+            if (method.Parent is null)
+                return;
+
+            var methodParameterType = context.SemanticModel.GetTypeInfo(parameterType, cancellationToken).Type;
+            if (methodParameterType is null)
+                return;
+
+            var siblingMethods = method.Parent.ChildNodes().OfType<MethodDeclarationSyntax>().Where(m => m != method && m.Identifier.Text == method.Identifier.Text);
+            if (!siblingMethods.Any())
+                return;
+
+            foreach (var sibling in siblingMethods)
+            {
+                foreach (var siblingParameter in sibling.ParameterList.Parameters)
+                {
+                    if (siblingParameter.Type is null)
+                        continue;
+
+                    var siblingParameterType = context.SemanticModel.GetTypeInfo(siblingParameter.Type, cancellationToken).Type;
```

Comment:
💭 This is a bit worrisome. `GetTypeInfo` can be really slow. If we switch to using the semantic model for finding overloads I think this problem will go away automatically.